spyder/plugins/pylint/widgets/pylintgui.py

```diff
@@ -312,6 +312,14 @@ def show_log(self):
 
     @Slot()
     def start(self):
+        """ Start the code analysis. """
+        # Check if the file should be automatically saved before analysis
+        if self.parent and self.parent.get_option('save_before', True):
+            # Try saving it
+            if not self.parent.main.editor.save():
+                # Saving failed for some reason, do not analyze
+                return

```

Comment:
This is not the way we handle communications among plugins (in this case Pylint and the Editor). Our design is the following:

* Widgets must emit a Qt signal when they need an action to be performed in another plugin.
* The plugin that owns the widget connects a slot to that signal and in that slot asks another plugin to perform the corresponding action.

So, in this case you need to emit a signal here, connect to it in `pylint/plugin.py`, and define its corresponding slot so that it calls `editor.save()`.

For an example of how that's done in another plugins, please see the `register_plugin` method of 

https://github.com/spyder-ide/spyder/blob/master/spyder/plugins/explorer/plugin.py

Reply:
OK! Will do! Sorry about that.

Reply:
No worries. That's not documented so it's hard to guess by looking at the code.